### ISSUE TITLE
Tracing doesn't work after upgrade to .NET 9 / Serilog 4.2.0

### ISSUE THREAD

**Issue by Wombatush** [serilog-tracing/serilog-tracing#168]
I have recently upgraded my sample project to .NET 9 (previously .NET 8); As part of the upgrade I had to upgrade Serilog.Extensions.Hosting package to the version 9.0.0 which in turn demanded Serilog 4.2.0 as a minimum.

After the upgrade the behavior has changed in a way, where StartActivity() no longer prints anything after disposal. It appears that the LoggerActivity, regardless of my reconfig attempts is always null (and I also checked it is not null in .NET 8 version / Serilog 4.0.1).

The sample project reproducing the issue is here: https://github.com/Wombatush/serilog-tracing-issue - a stripped down version of it; in short, I have an http-triggered function that supposed to log the activity ID, which is always null.

Could you please advise have I missed something in the config, or is there an issue with .NET 9 version? Thank you.

**Comment by nblumhardt** [comment]
Hi @Wombatush, thanks for the note.

The .NET 9 changes in Serilog.Extensions.Hosting and Serilog are fairly minimal and already in wide use so I'd bet on this being a Functions-specific issue.

Azure Functions can be quite tricky to deal with; I'm not sure where I'd start with this one. If you convert the project to a regular non-Functions EXE do you still see problems?

**Comment by Wombatush** [comment]
@nblumhardt yes, when executed as a console app, everything works fine, so it is indeed Functions-specific issue. 

Not too sure either, should I raise this to MS team?

**Comment by Wombatush** [comment]
@nblumhardt I think I managed to make it work, so will post here what I've found.

After extensive search I found this https://learn.microsoft.com/en-us/azure/azure-functions/opentelemetry-howto?tabs=app-insights&pivots=programming-language-csharp

The above says "OpenTelemetry support for Azure Functions is currently in preview.", however when I added "telemetryMode": "openTelemetry" to my host.json file, the sample application did work.

![image](https://github.com/user-attachments/assets/b9dcab94-d619-460f-ae96-33c6d320be1e)

**Comment by nblumhardt** [comment]
That's interesting - thanks for following up. _SerilogTracing_ doesn't rely on anything in OpenTelemetry, but both will be using the same .NET diagnostic APIs. My guess is that earlier Functions versions use/set these up in ways that are incompatible with modern usage, and the `"openTelemetry"` option there "un-breaks" them.

Good to know; I'm keen to hear any more info that turns up about this over time.

**Comment by Wombatush** [comment]
@nblumhardt It relies on Activity (and ActivitySource), which _perhaps_ in turn relies on OpenTelemetry in some way, at least in Functions. After all, that was an issue I was facing: Activity was null in LoggerActivity, and nothing was printed in Serilog / Seq.

**Comment by KodrAus** [comment]
Just coming back through some triage. I don't think there's anything for us to do here in SerilogTracing. There's a configuration option in Azure Functions that's just been given the name _openTelemetry_ to make it compatible with the OpenTelemetry SDK, which also happens to be compatible with SerilogTracing.

I'll go ahead and close this one, but it'll remain as a searchable record for anyone else in Azure Functions that hits this issue.
